src/tests/users/signup.test.js

```diff
@@ -1,30 +1,268 @@
 const expect = require('expect')
 const { request } = require('../../utils/test')
-const { UserFactory } = require('../../utils/factories/')
+const { UserFactory, QualificationFactory } = require('../../utils/factories/')
+const Qualification = require('../../models/qualifications')
+const User = require('../../models/user')
 
 describe('signup', () => {
   const fakeUser = UserFactory.generate()
+  const fakeUser2 = UserFactory.generate()
+  let users = []
+  let skills = []
+  let knowledge = []
+  let user = null
+  before(async () => {
+    for (let y = 0; y < 5; y++) {
+      const skill = await Qualification.create(
+        QualificationFactory.generateSkill()
+      )
+      skills.push(`"${skill._id}"`)
+      const knowledge1 = await Qualification.create(
+        QualificationFactory.generateKnowledge()
+      )
+      knowledge.push(`"${knowledge1._id}"`)
+    }
+    for (let x = 0; x <= 1; x++) {
+      let user = await UserFactory.generate()
+      user = await User.signup(user)
+      users.push(user)
+    }
+  })
+
+  it("should not signup if mentor is not certified or mentor's email is not verified", () => {
+    return request({
+      query: `
+        mutation {
+          signup(record: {
+              firstName: "${fakeUser.firstName}"
+              lastName: "${fakeUser.lastName}"
+              email: "${fakeUser.email}"
+              password: "${fakeUser.password}"  
+              username: "${fakeUser.username}"
+              mentor: "${users[0]._id}"
+              skills: [${skills}]
+              knowledge: [${knowledge}]
+          }) {
+            record {
+              email
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('errors')
+        expect(Array.isArray(res.body.errors)).toBe(true)
+      })
+      .expect(200)
+  })
+
+  it('should make user a mentor', () => {
```

Comment:
Can we just certify a user to be a mentor like this? Is there any business logic involved?

Cos this test doesn't say anything rather than merely updating a field

Reply:
well user have to be a mentor in order for being selected as a mentor during singup. It is making dummy user mentor only to check whether he appears in mentors list after becoming mentor.